src/fastfield/delete.rs

```diff
@@ -38,6 +37,34 @@ pub fn write_delete_bitset(
     Ok(())
 }
 
+/// Merges two `DeleteBitSet` into a new one.
+pub fn merge_delete_bitset(left: &DeleteBitSet, right: &DeleteBitSet) -> DeleteBitSet {
+    let left_data = left.data.as_slice();
+    let right_data = right.data.as_slice();
+
+    let mut merged = vec![];
+    merged.resize(left_data.len().max(right_data.len()), 0);
+
+    for (merged_el, left_el) in merged.iter_mut().zip(left_data.iter()) {
+        *merged_el = *left_el;
+    }
+
+    for (merged_el, right_el) in merged.iter_mut().zip(right_data.iter()) {
+        *merged_el |= *right_el;
+    }
+
+    let num_deleted: usize = merged
```

Comment:
There is a from_slice in DeleteBitset.

You could add `from_bytes` and factorize this.